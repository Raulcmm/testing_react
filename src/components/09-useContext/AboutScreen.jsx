import React, { useContext } from "react";
import { userContext } from "./userContext";

const AboutScreen = () => {
	const { user, setUser } = useContext(userContext);
	const handleClick = () => {
		setUser({});
	};

	return (
		<div>
			<h1>AboutScreen</h1>
			<hr />
			<pre>{JSON.stringify(user, null, 3)}</pre>
			<h2>{user.name}</h2>

			<button className="btn btn-warning" onClick={handleClick}>
				delete data
			</button>
		</div>
	);
};

export default AboutScreen;
