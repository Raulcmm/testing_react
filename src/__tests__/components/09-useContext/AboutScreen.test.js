import { act } from "@testing-library/react-hooks";
import { mount, shallow } from "enzyme";
import React from "react";
import AboutScreen from "../../../components/09-useContext/AboutScreen";
import UserContextProvider, {
	userContext,
} from "../../../components/09-useContext/userContext";

describe("Tests to AboutScreen", () => {
	const wrapper = mount(
		<UserContextProvider>
			<AboutScreen />
		</UserContextProvider>
	);

	test("Debe  hacer match", () => {
		expect(wrapper).toMatchSnapshot();
	});

	test("debe mostrar la data del usuario", () => {
		const pre = wrapper.find("pre").text().trim().includes("Raul Camacho");
		expect(pre).toBe(true);
	});

	test("Debe limpiar la informacion ", () => {
		wrapper.find("button").simulate("click");
		expect(wrapper.find("h2").text().trim()).toBe("");
	});
});
