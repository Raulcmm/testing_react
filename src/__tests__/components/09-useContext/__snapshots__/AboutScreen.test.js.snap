// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests to AboutScreen Debe  hacer match 1`] = `
<div>
  <h1>
    AboutScreen
  </h1>
  <hr />
  <pre>
    {
   "id": 1,
   "name": "Raul Camacho"
}
  </pre>
  <h2>
    Raul Camacho
  </h2>
  <button
    className="btn btn-warning"
    onClick={[Function]}
  >
    delete data
  </button>
</div>
`;
